doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1257,115 +1257,87 @@ Managing the local context
    for any unresolved variables rather than failing.  Typically this happens when
    using a ``%`` intropattern (see :n:`@simple_intropattern`).
 
-.. tacn:: clear @ident
-   :name: clear
+.. tacn:: clear {? {? - } {+ @ident } }
 
-   This tactic erases the hypothesis named :n:`@ident` in the local context of
-   the current goal. As a consequence, :n:`@ident` is no more displayed and no
-   more usable in the proof development.
+   clear
+   clear ident ...
+   clear - ident ...
 
-   .. exn:: No such hypothesis.
-      :undocumented:
+   Erases hypotheses from the context of the current goal, provided that the
+   remaining hypotheses and the goal don't depend directly or indirectly on the
+   erased hypotheses.
+   That means the hypotheses will no longer appear in the context and therefore
+   can't be used in subsequent proof steps.
+
+   :n:`clear`
+     All hypotheses are erased.  This may leave the context unchanged; this form never fails.
+
+   :n:`clear {+ @ident }`
+     Erases the named hypotheses.
+
+   :n:`clear - {+ @ident }`
+     Erases all hypotheses that are not named by(ass) the :n:`@ident`\s.
 
    .. exn:: @ident is used in the conclusion.
       :undocumented:
 
    .. exn:: @ident is used in the hypothesis @ident.
       :undocumented:
 
-   .. tacv:: clear {+ @ident}
-
-      This is equivalent to :n:`clear @ident. ... clear @ident.`
-
-   .. tacv:: clear - {+ @ident}
-
-      This variant clears all the hypotheses except the ones depending in the
-      hypotheses named :n:`{+ @ident}` and in the goal.
+   .. tacn:: clearbody {+ @ident }
 
-   .. tacv:: clear
-
-      This variants clears all the hypotheses except the ones the goal depends on.
-
-   .. tacv:: clear dependent @ident
```

Comment:
Where has the documentation of `clear dependent` gone?

Reply:
I couldn't find a definition in the mlgs and it's an Ltac definition.  Where is the tactic defined?

Reply:
Here: https://github.com/coq/coq/blob/e72849358f760f55b4dea5311d8ea63e3f90f825/theories/Init/Tactics.v#L232